### PR TITLE
feat(UI): Display widgets in dashboard

### PR DESCRIPTION
**Pull request by Thebarda**
## Description

This implements a dashboard layout system, adds sample widgets and adds widgets build process to the CI.
Video will come later

This implements this ticket: [MON-18490](https://centreon.atlassian.net/browse/MON-18490)

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 21.10.x
- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [x] 23.10.x (master)

<h2> How this pull request can be tested ? </h2>

#### Community contributors & Centreon team

- [x] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

[MON-18490]: https://centreon.atlassian.net/browse/MON-18490?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: centreon/www/front_src/src/Dashboards/Dashboard/Layout/index.tsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+import { Panel } from '../models';
+
+import DashboardPanel from './Panel/Panel';
+import PanelHeader from './Panel/PanelHeader';
+
+const Layout = (): JSX.Element => {
+  const [dashboard, setDashboard] = useAtom(dashboardAtom);
+  const isEditing = useAtomValue(isEditingAtom);
+
+  const changeLayout = (layout: Array<Layout>): void => {
```
[the remark this answers is not in the document]
Comment (Thebarda) [reply centreon/www/front_src/src/Dashboards/Dashboard/Layout/index.tsx:21]: Why?

File: centreon/www/front_src/src/Dashboards/Dashboard/Layout/index.tsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+      layout={dashboard.layout}
+    >
+      {dashboard.layout.map(({ i }) => {
+        return (
+          <DashboardItem
+            header={isEditing ? <PanelHeader id={i} /> : undefined}
+            key={i}
+          >
+            <DashboardPanel id={i} />
+          </DashboardItem>
```
[the remark this answers is not in the document]
Comment (Thebarda) [reply centreon/www/front_src/src/Dashboards/Dashboard/Layout/index.tsx:55]: I don't link the dot naming convention

File: centreon/package.json [hunk 3]
```diff
@@ -3,12 +3,20 @@
...
+    "centreon:build:analyze": "webpack --mode production --config ./webpack.config.analyze.js",
+    "centreon:build:dev": "cross-env WEBPACK_ENV=development webpack serve --mode development --config ./webpack.config.dev.js",
+    "centreon:watch": "cross-env WEBPACK_ENV=serve webpack serve --mode development --config ./webpack.config.dev.js",
+    "build": "pnpm widgets:build && pnpm centreon:build",
+    "build:dev": "pnpm widgets:build:dev && pnpm centreon:build:dev",
+    "build:watch": "pnpm centreon:watch & pnpm widgets:watch",
+    "build:analyze": "pnpm widgets:build:analyze && pnpm centreon:build:analyze",
     "eslint": "eslint ./www/front_src --ext .js,.jsx,.ts,.tsx --max-warnings 0",
-    "eslint:fix": "pnpm eslint --fix",
+    "eslint:fix": "pnpm centreon:eslint --fix",
```
[the remark this answers is not in the document]
Comment (Thebarda) [reply centreon/package.json:19]: Go to the widgets formder then use the lint command

File: centreon/packages/ui/src/Dashboard/index.stories.tsx
[the remark this answers is not in the document]
Comment (Thebarda) [reply centreon/packages/ui/src/Dashboard/index.stories.tsx:1]: Yes

**Review by abuathier — APPROVED** [review]

**Pull request closed without merging**